src/unistd.rs

```diff
@@ -2778,3 +2778,42 @@ impl Group {
         })
     }
 }
+
+/// Get the name of the terminal device that is open on file descriptor fd
+/// (see [`ttyname(3)`](http://man7.org/linux/man-pages/man3/ttyname.3.html)).
+///
+/// # Safety
+///
+/// `ttyname()` mutates global variables and is *not* threadsafe.
+/// Mutating global variables is always considered `unsafe` by Rust and this
+/// function is marked as `unsafe` to reflect that.
+///
+/// For a threadsafe and non-`unsafe` alternative, see `ttyname_r()`.
+#[inline]
+pub unsafe fn ttyname(fd: RawFd) -> Result<String> {
```

Comment:
Is there a good reason why anybody would ever want `ttyname`?  Since the invention of `ttyname_r`, I think the latter should be used in all cases.  If you don't know of a use case where `ttyname_r` does not suffice, then I'd just as soon we don't bind `ttyname` at all.

Reply:
yeah, good point - i was mainly copying from the `ptsname` interface, but i agree that it's really not necessary.